src/segments/nba.go

```diff
@@ -0,0 +1,438 @@
+)
+
+// segment struct, makes templating easier
+type NBA struct {
+	props properties.Properties
+	env   platform.Environment
+
+	NBAData
+}
+
+// NBA struct contains parsed API data that care about for the segment
+type NBAData struct {
+	HomeTeam       string
+	AwayTeam       string
+	Time           string
+	GameDate       string
+	StartTimeUTC   string
+	GameStatus     GameStatus // 1 = scheduled, 2 = in progress, 3 = finished
+	HomeScore      int
+	AwayScore      int
+	HomeTeamWins   int
+	HomeTeamLosses int
+	AwayTeamWins   int
+	AwayTeamLosses int
+}
+
+const (
+	NBASeason        properties.Property = "season"
+	TeamName         properties.Property = "team"
+	DaysOffset       properties.Property = "days_offset"
+	NBAScoreURL      string              = "https://cdn.nba.com/static/json/liveData/scoreboard/todaysScoreboard_00.json"
+	NBASchedURLPart1 string              = "https://stats.nba.com/stats/internationalbroadcasterschedule?LeagueID=00&Season="
+	NBASchedURLPart2 string              = "&RegionID=1&EST=Y"
+)
+
+// Custom type for GameStatus
+type GameStatus int
+
+// Constants for GameStatus values
+const (
+	Scheduled  GameStatus = 1
+	InProgress GameStatus = 2
+	Finished   GameStatus = 3
+	NotFound   GameStatus = 4
+)
+
+// Int() method for GameStatus to get its integer representation
+// This is a helpful method if people want to come up with their own templates
+func (gs GameStatus) Int() int {
+	return int(gs)
+}
+
+// String() method for GameStatus to get its string representation
+func (gs GameStatus) String() string {
+	switch gs {
+	case Scheduled:
+		return "Scheduled"
+	case InProgress:
+		return "In Progress"
+	case Finished:
+		return "Finished"
+	case NotFound:
+		return "Not Found"
+	default:
+		return "Unknown"
+	}
+}
+
+// All of the structs needed to retrieve data from the live score endpoint
+type ScoreboardResponse struct {
+	Scoreboard Scoreboard `json:"scoreboard"`
+}
+
+type Scoreboard struct {
+	GameDate string `json:"gameDate"`
+	Games    []Game `json:"games"`
+}
+
+type Game struct {
+	GameStatus     int    `json:"gameStatus"`
+	GameStatusText string `json:"gameStatusText"`
+	GameTimeUTC    string `json:"gameTimeUTC"`
+	HomeTeam       Team   `json:"homeTeam"`
+	AwayTeam       Team   `json:"awayTeam"`
+}
+
+type Team struct {
+	TeamTricode string `json:"teamTricode"`
+	Wins        int    `json:"wins"`
+	Losses      int    `json:"losses"`
+	Score       int    `json:"score"`
+}
+
+// All the structs needed to get data from the schedule endpoint
+type ScheduleResponse struct {
+	ResultSets []ResultSet `json:"resultSets"`
+}
+
+type ResultSet struct {
+	CompleteGameList []ScheduledGame `json:"CompleteGameList,omitempty"`
+}
+
+type ScheduledGame struct {
+	VtAbbreviation string `json:"vtAbbreviation"`
+	HtAbbreviation string `json:"htAbbreviation"`
+	Date           string `json:"date"`
+	Time           string `json:"time"`
+}
+
+func (nba *NBA) Template() string {
```

Comment:
We usually only have a single template. Now, I get that this could be a bit too challenging in this use-case so I'm down to going this direction. But, then we would have to allow people to override the templates below, and I'd rather have that logic in 1 function instead of having it split across multiple. This logic `if nba.HomeTeamWins == 0 && nba.HomeTeamLosses == 0 && nba.AwayTeamWins == 0 && nba.AwayTeamLosses == 0` can be a boolean function on the nba struct, but I would add every other template directly in the `Template` function AND, if the user has its own `template` property, we simply return that one first without running any additional logic.

Reply:
I'm a bit confused. I was under the impression that a user can still have whatever template they'd like to use and override the template even in the way it is currently constructed. For example, I'm finding that something like this:

"segments": [
        {
          "background": "#e9ac2f",
          "foreground": "#8748dc",
          "leading_diamond": "\ue0b6",
          "style": "diamond",
	      "template": "Test this out {{ .HomeTeam }}",
          "trailing_diamond": "\ue0b0",
          "type": "nba",
          "properties": {
            "team":"LAL",
            "http_timeout": 1500,
	    "season": "2023",
	    "days_offset": 8
          }
        }
      ],
      "type": "prompt"
    }
  ],

Will render the template provided by the user "Test this out {{ .HomeTeam }}" and not the template that is being returned in Template(). 

As always, thanks for the help and taking the time to teach me what's going on here!

Reply:
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/98557455/c3ecff6f-113e-47df-a64f-d10cfe35b7e4)

Reply:
You're right, it's defined in the engine to do that. But, I would still export the included partial templates for override so the user can decide to override one or more without having to resort to a full override.

Reply:
Can you please explain how I should handle this? What do you mean by export the included partial templates?